clang/lib/Sema/SemaExpr.cpp

```diff
@@ -11773,7 +11786,25 @@ QualType Sema::CheckShiftOperands(ExprResult &LHS, ExprResult &RHS,
       isScopedEnumerationType(RHSType)) {
     return InvalidOperands(Loc, LHS, RHS);
   }
-  DiagnoseBadShiftValues(*this, LHS, RHS, Loc, Opc, LHSType);
+
+  BadShiftValueKind BSVKind =
+      DiagnoseBadShiftValues(*this, LHS, RHS, Loc, Opc, LHSType);
+  if (ExprEvalContexts.back().isConstantEvaluated()) {
```

Comment:
This feels pretty hacky but perhaps I am not fully understanding the solution.

Can you update the description with more details on why this is a good approach to the problem.

Reply:
This solution originally was posted on [Phabricator](https://reviews.llvm.org/D141192). I reviewed commit and found that it makes more sense to diagnose bad shift values inside function DiagnoseBadShiftValues(...) instead in mentioned IF. If there are no more questions, I will edit this part of code.